apm-agent-plugins/apm-logging-plugin/apm-logging-plugin-common/src/main/java/co/elastic/apm/agent/loginstr/reformatting/AbstractEcsReformattingHelper.java

```diff
@@ -439,6 +444,18 @@ private String getServiceName() {
         return configuredServiceName;
     }
 
+    /**
+     * We currently get the same service version that is reported in the metadata document.
+     * This may mismatch automatically-discovered service version (if not configured). However, we only set it
+     * once when configuring our appender, so we can have only one service version.
```

Comment:
This is because the auto-detection for different web apps happens after their logging setup is initialized, right?

[minor, feel free to ignore] I think the auto-detected version from a standalone jar (`META-INF/MANIFEST.mf`) would still be applied, right? Therefore, the variable `configuredServiceVersion` may be misleading as it could also be auto-discovered. Maybe rename to `globalServiceVersion`. But I see the same also applies to `configuredServiceName` so probably out of scope for this PR.

Reply:
Yeah, I agree. IIRC, it was really only the configured value when added and changed later to globally-auto-detected.
I'll change.